devops/providers/azure-devops/templates/infrastructure/azure-pipeline-build-stage.yml

```diff
@@ -83,6 +94,14 @@ jobs:
         TF_VAR_remote_state_account: $(REMOTE_STATE_ACCOUNT)
         TF_VAR_remote_state_container: $(REMOTE_STATE_CONTAINER)
 
+    - task: Bash@3
+      name: TeardownGitCredentialInjection
+      displayName: Reset Git configuration
+      condition: and(always(), eq(${{ coalesce(config.overrideGitPAT, 'false') }}, 'true'))
```

Comment:
I think that we *always* want to run this if the override for PAT is set, but I don't think the condition matches this. I think you want something like:

```yml
  ${{ if coalesce(config.overrideGitPAT, 'false') }}:
    - task: Bash@3
      name: TeardownGitCredentialInjection
      displayName: Reset Git configuration
      condition: always()
      inputs:
        targetType: 'inline'
        script: '[[ -z $(TEMPLATE_REPO_PAT) ]] && echo "No PAT provided; skipping." || git config --global --unset url."https://cobalt:$(TEMPLATE_REPO_PAT)@".insteadOf'

```

Reply:
As I understand the docs, the *always()* means that it will run even if there's a failure in the pipeline (which is what we would want), but... only if we actually set it. Therefore, the *and* allows us to say if we used a Git credential PAT, always undo that step -- even if there was a pipeline failure, cancellation, or some other issue. If there wasn't a Git credential PAT, you can skip it.